src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1508,28 +1557,199 @@ abstract void onApplication(
+                        payload.buffer(), payload.offset(), payload.sizeof(), extension);
+            }
+
+            private void doInitialEnd(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                if (!MqttState.initialOpened(state) || initialSlot != NO_SLOT)
+                {
+                    state = MqttState.closingInitial(state);
+                }
+                else
+                {
+                    flushInitialEnd(traceId, authorization, extension);
+                }
+            }
+
+            private void doInitialAbort(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                setInitialClosed();
+
+                doAbort(application, routeId, initialId, traceId, authorization, extension);
+            }
+
+            private void doInitialAbortIfNecessary(
+                long traceId,
+                long authorization)
+            {
+                if (!MqttState.initialClosed(state))
+                {
+                    doInitialAbort(traceId, authorization, EMPTY_OCTETS);
+                }
+            }
+
+            private void flushInitialEnd(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                setInitialClosed();
+                doEnd(application, routeId, initialId, traceId, authorization, extension);
+            }
+
+            private void setInitialClosed()
+            {
+                assert !MqttState.initialClosed(state);
+
+                state = MqttState.closeInitial(state);
+                cleanupInitialSlotIfNecessary();
+
+                if (MqttState.closed(state))
+                {
+                    publishers.remove(topicKey(topicFilter));
+                    activePublishers.computeIfPresent(topicFilter, (key, value) -> value--);
+                }
+            }
+
+            @Override
+            public void onApplicationInitial(
+                int msgTypeId,
+                DirectBuffer buffer,
+                int index,
+                int length)
+            {
+                switch (msgTypeId)
+                {
+                case WindowFW.TYPE_ID:
+                    final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                    onWindowInitial(window);
+                    break;
+                case ResetFW.TYPE_ID:
+                    final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                    onResetInitial(reset);
+                    break;
+                case SignalFW.TYPE_ID:
+                    final SignalFW signal = signalRO.wrap(buffer, index, index + length);
+                    onSignalInitial(signal);
+                    break;
+                }
```

Comment:
Suggest renaming these methods to `onApplicationWindow`, `onApplicationReset` and `onApplicationSignal` instead.